File: DSA/Priyanshi/Day16_que2.java
class Node {
	int data;
	Node left, right;

	Node(int v)
	{
		data = v;
		left = right = null;
	}
}

class Day16_que2 {

	public static void printInorder(Node node)
	{
		if (node == null)
			return;

		printInorder(node.left);

		System.out.print(node.data + " ");

		printInorder(node.right);
	}

	public static void main(String[] args)
	{
		Node root = new Node(1);
		root.left = new Node(2);
		root.right = new Node(3);
		root.left.left = new Node(4);
		root.left.right = new Node(5);

		System.out.println("Inorder traversal of binary tree is: ");
		printInorder(root);
	}
}

File: DSA/Devansh-Nishchal-Mandloi/day16que2.cpp
#include <iostream>
#include <vector>
using namespace std;

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

// Function to perform inorder traversal.
void inorderTraversal(TreeNode* root, vector<int>& result) {
    if (root == NULL) {
        return;
    }
    inorderTraversal(root->left, result);
    result.push_back(root->val);
    inorderTraversal(root->right, result);
}

// Example usage
int main() {
    // Creating a simple binary tree
    TreeNode* root = new TreeNode(1);
    root->left = new TreeNode(2);
    root->right = new TreeNode(3);
    root->left->left = new TreeNode(4);
    root->left->right = new TreeNode(5);

    vector<int> result;
    inorderTraversal(root, result);
    for (int val : result) {
        cout << val << " ";
    }
    cout << endl;

    return 0;
}
